src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,67 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int STORAGE_MAX_CAPACITY = 10;
+    private int size = 0;
+    private final StorageElement<K, V>[] storageElements = new StorageElement[STORAGE_MAX_CAPACITY];
+
     @Override
     public void put(K key, V value) {
+        StorageElement<K, V> storageElement = new StorageElement<>(key, value);
+        for (int i = 0; i < STORAGE_MAX_CAPACITY; i++) {
+            if (storageElements[i] == null) {
+                size++;
+                storageElements[i] = storageElement;
+                return;
+            } else if (storageElements[i].isEmpty()
+                    || storageElements[i].hasKey(storageElement.getKey())) {
+                storageElements[i] = storageElement;
+                return;
+            }
+        }
+        throw new RuntimeException("StorageImpl has no more space.");
     }
 
     @Override
     public V get(K key) {
+        for (StorageElement<K, V> storageElement : storageElements) {
+            if (storageElement != null && !storageElement.isEmpty() && storageElement.hasKey(key)) {
+                return (V) storageElement.value;
+            }
+        }
         return null;
     }
 
     @Override
     public int size() {
-        return -1;
+        return this.size;
```

Comment:
this is redundant here, there are no other `size` named variables in scope of that method
```suggestion
        return size;
```